### PR TITLE
Create a 502 error page to automatically refresh if backend status changes

### PR DESCRIPTION
**Pull request by alexymik**
If a container goes down and the gateway timeout returns a 502, periodically check the backend using a HEAD request. If the 502 is resolved (container reboots, etc.) then automatically refresh the page. Also uses a backoff to increase the time between checks.

### CODE REVIEW

File: plugins/nginx-vhosts/templates/502-error.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    var retry_limit = 4096; // Maximum backoff time, ~68 minutes
+
+    check_gateway = function () {
+
+      var request = new XMLHttpRequest();
+      request.open('HEAD', window.location.href, true);
+
+      request.onload = function () {
+
+        console.dir(window.location.href, 'returned', request.status);
```
Comment (josegonzalez) [inline plugins/nginx-vhosts/templates/502-error.html]: Can you remove the console log calls?

Comment (alexymik) [reply plugins/nginx-vhosts/templates/502-error.html]: Done

**Comment by alexymik** [comment]
The page title could be renamed to something else and the return status could be modified to check for only 502. I had it in the 500-error.html originally

File: plugins/nginx-vhosts/templates/502-error.html (hunk 1 above)
Comment (josegonzalez) [reply plugins/nginx-vhosts/templates/502-error.html]: Missed one

Comment (alexymik) [reply plugins/nginx-vhosts/templates/502-error.html]: Oops, sorry

File: plugins/nginx-vhosts/templates/502-error.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>We're sorry, but something went wrong (500)</title>
```
Comment (josegonzalez) [inline plugins/nginx-vhosts/templates/502-error.html]: Should say 502 here (just to distinguish it since we can).

**Pull request merged**